### ISSUE TITLE
In partialFC pyTorch version, loading pretrained model weights does not make any sense, what's wrong with my code ?

### ISSUE THREAD

**Issue by CatLoves** [deepinsight/insightface#1415]
Hi everyone :

I want to use the pretrained model weights of partial_fc_glint360k_r100 from 16_backbone.pth, then finetuning the backbone with my own dataset. 

The strange problem is that:  I can load model weights from 16_backbone.pth, but the initial loss is the same as that of not using pretrained model weights, in other words,  loading pretrained model weights does not make any sense.

There may be some problem with my code, so I paste my code here, can anyone give me some suggestions ?

Core code (partial_fc.py):

`

def main(local_rank):

    """  
    主函数

    Args:
        local_rank: list[integer], rank of the process on the local machine, e.g., [0, 1, 2, 3] if there are 4 GPUs in this machine

    Comments:  local_rank vs rank
                    |    Node1  |   Node2    |
        ____________| p1 |  p2  |  p3  |  p4 |
        local_rank  | 0  |   1  |  0   |   1 |
        rank        | 0  |   1  |  2   |   4 |
    """

    """ 设置一些与分布式训练相关的参数 """
    dist.init_process_group(backend='nccl', init_method='env://')
    cfg.local_rank = local_rank
    torch.cuda.set_device(local_rank)
    cfg.rank = dist.get_rank()
    cfg.world_size = dist.get_world_size()
    # debug
    print(f'cfg.local_rank:')
    print(cfg.local_rank)

    """ 构建训练集的 data loader """
    trainset = MXFaceDataset(
        root_dir=cfg.rec, 
        local_rank=local_rank
    )
    train_sampler = torch.utils.data.distributed.DistributedSampler(
        trainset, shuffle=True)
    train_loader = DataLoaderX(local_rank=local_rank,
                               dataset=trainset,
                               batch_size=cfg.batch_size,
                               sampler=train_sampler,
                               num_workers=0,
                               pin_memory=True,
                               drop_last=False)

    """ 构建网络 backbone """
    backbone = backbones.iresnet100(pretrained=False).to(local_rank)
    backbone.train()

    """ 是否使用预训练的模型权重 """
    if cfg.use_pretrained:

        backbone.load_state_dict(torch.load(cfg.pretrained_model_path))
        # show message
        logger.info(f'\n Loaded pretrained model weights from {cfg.pretrained_model_path} \n')

    # Broadcast init parameters
    for ps in backbone.parameters():
        dist.broadcast(ps, 0)

    """ 
    做两件事情：
        1. 将 backbone 复制多份，每个设备一份；
        2. 将 input data 划分多份，每个设备一份
    """
    backbone = torch.nn.parallel.DistributedDataParallel(
        module=backbone,
        broadcast_buffers=False,
        device_ids=[cfg.local_rank])
    backbone.train()

    # Memory classifer
    dist_sample_classifer = DistSampleClassifier(
        rank=dist.get_rank(),
        local_rank=local_rank,
        world_size=cfg.world_size)

`
Other codes are not modified.

Any suggestions would be appreciated !

**Comment by anxiangsir** [comment]
you should also load partial fc parameters like this:
```python
self.weight_name = os.path.join(self.prefix, "rank:{}_softmax_weight.pt".format(self.rank))
try:
    self.weight: torch.Tensor = torch.load(self.weight_name)
except (FileNotFoundError, KeyError, IndexError):
    self.weight = torch.normal(0, 0.01, (self.num_local, self.embedding_size), device=self.device)
```

**Comment by anxiangsir** [comment]
partial-fc pytorch has been merged into arcface_torch
https://github.com/deepinsight/insightface/tree/master/recognition/arcface_torch

**Comment by staystay** [comment]
> you should also load partial fc parameters like this:
> 
> ```python
> self.weight_name = os.path.join(self.prefix, "rank:{}_softmax_weight.pt".format(self.rank))
> try:
>     self.weight: torch.Tensor = torch.load(self.weight_name)
> except (FileNotFoundError, KeyError, IndexError):
>     self.weight = torch.normal(0, 0.01, (self.num_local, self.embedding_size), device=self.device)
> ```
> 
> partial fc parameters also should be saved.
> 
> ```python
> def save_params(self):
>     torch.save(self.weight.data, self.weight_name)
> ```

请问下，作者开源的是在8卡上训练的，但是我机器只有4卡，那我load weight的时候，正常来说只能load 4个，那是不是要在作者开源的模型上微调，一定要用8卡的呀

**Comment by staystay** [comment]
> you should also load partial fc parameters like this:
> 
> ```python
> self.weight_name = os.path.join(self.prefix, "rank:{}_softmax_weight.pt".format(self.rank))
> try:
>     self.weight: torch.Tensor = torch.load(self.weight_name)
> except (FileNotFoundError, KeyError, IndexError):
>     self.weight = torch.normal(0, 0.01, (self.num_local, self.embedding_size), device=self.device)
> ```
> 
> partial fc parameters also should be saved.
> 
> ```python
> def save_params(self):
>     torch.save(self.weight.data, self.weight_name)
> ```

请问下，作者开源的是在8卡上训练的，但是我机器只有4卡，那我load weight的时候，正常来说只能load 4个，那是不是要在作者开源的模型上微调，一定要用8卡的呀

**Comment by staystay** [comment]
> you should also load partial fc parameters like this:
> 
> ```python
> self.weight_name = os.path.join(self.prefix, "rank:{}_softmax_weight.pt".format(self.rank))
> try:
>     self.weight: torch.Tensor = torch.load(self.weight_name)
> except (FileNotFoundError, KeyError, IndexError):
>     self.weight = torch.normal(0, 0.01, (self.num_local, self.embedding_size), device=self.device)
> ```
> 
> partial fc parameters also should be saved.
> 
> ```python
> def save_params(self):
>     torch.save(self.weight.data, self.weight_name)
> ```

请问下，作者开源的是在8卡上训练的，但是我机器只有4卡，那我load weight的时候，正常来说只能load 4个，那是不是要在作者开源的模型上微调，一定要用8卡的呀
